data-structures/ruby/doubly_linked_list/empty.rb

```diff
@@ -0,0 +1,11 @@
+module DoublyLinkedList
+  class Empty
```

Comment:
Isn't `Empty` a refined type of `Node`?  That should push you in the direction of making `Empty` a subclass of `Node`.